communitymanager/src/test/java/org/pesmypetcare/communitymanager/managers/ForumManagerClientTest.java

```diff
@@ -0,0 +1,198 @@
+        creator = "John";
+        token = "token";
+        groupNameEncoded = HttpParameter.encode(parentGroup);
+        forum = new ForumData(forumName, creator, null);
+        forumNameEncoded = HttpParameter.encode(forumName);
+        headers = new HashMap<>();
+        headers.put(token, token);
+        date = "2020-05-12";
+    }
+
+    @Test
+    public void createForum() throws MyPetCareException {
+        given(httpClient.post(anyString(), isNull(), isNull(), anyString())).willReturn(httpResponse);
+
+        client.createForum(parentGroup, forum);
+        verify(httpClient).post(eq(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup)), isNull(), isNull(),
+                eq(gson.toJson(forum)));
+    }
+
+    @Test
+    public void deleteForum() throws MyPetCareException {
+        HttpParameter[] params = new HttpParameter[1];
+        params[0] = new HttpParameter(FORUM_PARAMETER, forumName);
+        given(httpClient.delete(anyString(), any(HttpParameter[].class), isNull(), isNull())).willReturn(httpResponse);
+
+        client.deleteForum(parentGroup, forumName);
+        verify(httpClient)
+                .delete(eq(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup)), eq(params), isNull(), isNull());
+    }
+
+    @Test
+    public void getForum() throws MyPetCareException {
+        HttpParameter[] params = new HttpParameter[1];
+        params[0] = new HttpParameter(FORUM_PARAMETER, forumName);
+        given(httpClient
+                .get(eq(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup)), eq(params), isNull(), isNull()))
+                .willReturn(httpResponse);
+        given(httpResponse.asString()).willReturn(gson.toJson(forum));
+
+        ForumData forumData = client.getForum(parentGroup, forumName);
+        assertEquals("Should return the requested forum data.", forum, forumData);
+    }
+
+    @Test
+    public void getAllForums() throws MyPetCareException {
+        given(httpClient.get(eq(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup)), isNull(), isNull(), isNull()))
+                .willReturn(httpResponse);
+        List<ForumData> forums = new ArrayList<>();
+        forums.add(forum);
+        given(httpResponse.asString()).willReturn(gson.toJson(forums));
+        List<ForumData> response = client.getAllForums(parentGroup);
+        assertEquals("Should return all the existing forums in a group.", forums, response);
+    }
+
+    @Test
+    public void updateName() throws MyPetCareException {
+        HttpParameter[] params = new HttpParameter[1];
+        String newName = "German Shepherds";
+        params[0] = new HttpParameter("newName", newName);
+        given(httpClient.put(anyString(), any(HttpParameter[].class), isNull(), isNull())).willReturn(httpResponse);
+
+        client.updateName(parentGroup, forumName, newName);
+        verify(httpClient).put(eq(COMMUNITY_BASE_URL + groupNameEncoded + "/" + forumNameEncoded), eq(params), isNull(),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "/" appears 6 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5538337)